libeos-updater-util/flatpak-util.c

```diff
@@ -2443,3 +2443,98 @@ eos_updater_get_flatpak_installation (GCancellable *cancellable, GError **error)
 
   return flatpak_installation_new_system (cancellable, error);
 }
+
+gboolean
+euu_flatpak_transaction_install (FlatpakInstallation *installation,
+                                 const gchar         *remote,
+                                 const gchar         *formatted_ref,
+                                 gboolean             no_deploy,
+                                 gboolean             no_pull,
+                                 GCancellable        *cancellable,
+                                 GError              **error)
+{
+  g_autoptr(FlatpakTransaction) transaction = NULL;
+
+  g_return_val_if_fail (FLATPAK_IS_INSTALLATION (installation), FALSE);
+  g_return_val_if_fail (formatted_ref != NULL, FALSE);
+  g_return_val_if_fail (!(no_deploy && no_pull), FALSE);
+  g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable), FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  transaction = flatpak_transaction_new_for_installation (installation, cancellable, error);
+  if (transaction == NULL)
+    return FALSE;
+
+  flatpak_transaction_set_no_deploy (transaction, no_deploy);
+  flatpak_transaction_set_no_pull (transaction, no_pull);
+
+  if (!flatpak_transaction_add_install (transaction,
+                                        remote,
+                                        formatted_ref,
+                                        NULL, /* subpaths */
+                                        error))
+    return FALSE;
+
+  return flatpak_transaction_run (transaction, cancellable, error);
+}
+
+gboolean
+euu_flatpak_transaction_update (FlatpakInstallation *installation,
+                                const gchar         *formatted_ref,
+                                gboolean             no_deploy,
+                                gboolean             no_pull,
+                                gboolean             no_prune,
```

Comment:
Yeah I think you're right, though using the existing flags is slightly an abuse because they are documented as being for `flatpak_installation_...` functions, e.g. `FlatpakInstallFlags`. But that doesn't seem important.

Reply:
Actually maybe it's not worth it, because for example taking a `FlatpakUninstallFlags` argument implies we support `FLATPAK_UNINSTALL_FLAGS_NO_TRIGGERS`, which we don't. We can document that limitation and add a `g_return_val_if_fail...` check for it, but it's hard to see how that confusion is better than the `FALSE, /* no_pull */` confusion.

Reply:
Or you could add a new flags type. Up to you :shrug:

Reply:
I think we should leave it as booleans in this case. If we add a new flags type we're left with a similar issue of not fully accepting the flags since the three functions (install, update, uninstall) each take a different subset of the flags. Maybe in the future if we want to use more of the flags we'd want to refactor here.